kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/symbol/kotlin/KSFunctionDeclarationImpl.kt

```diff
@@ -43,6 +43,9 @@ class KSFunctionDeclarationImpl private constructor(internal val ktFunctionSymbo
     }
 
     override val functionKind: FunctionKind by lazy {
+        if (modifiers.contains(Modifier.JAVA_STATIC)) {
+            FunctionKind.STATIC
+        }
```

Comment:
Thanks for the fix! Unfortunately getting modifiers can be slow in some cases. Can we patch the `when` case `KaSymbolLocation.CLASS` below, and make it consider `ktFunctionSymbol.isStatic` when it is a `KaNamedFunction`? I.e., something like
```
            KaSymbolLocation.CLASS -> {
                if (ktFunctionSymbol is KaNamedFunctionSymbol && ktFunctionSymbol.isStatic)
                    FunctionKind.STATIC
                else
                    FunctionKind.MEMBER
            }
```

Reply:
Yeah, looks way better, thank you!